public/app/core/logs_model.ts

```diff
@@ -245,23 +249,33 @@ export function getSeriesProperties(
   let visibleRange = absoluteRange;
   let resolutionIntervalMs = intervalMs;
   let bucketSize = Math.max(resolutionIntervalMs * pxPerBar, minimumBucketSize);
+  let timeSpanOfLogs;
+  let timeSpanOfRange;
   // Clamp time range to visible logs otherwise big parts of the graph might look empty
   if (absoluteRange) {
-    const earliest = sortedRows[0].timeEpochMs;
-    const latest = absoluteRange.to;
-    const visibleRangeMs = latest - earliest;
+    const earliestTsLogs = sortedRows[0].timeEpochMs;
+    const latestTsLogs = sortedRows[sortedRows.length - 1].timeEpochMs;
+    const earliestTsAbsolute = absoluteRange.from;
+    const latestTsAbsolute = absoluteRange.to;
+
+    // Calculate timespan for received logs and range. This is used to calculate timespan coverage meta info of received logs.
+    // We are rounding this as some ds are rounding range in their requests which resulted in incorrect coverage (in edge cases).
+    timeSpanOfLogs = Math.ceil((latestTsLogs - earliestTsLogs) / 1000);
+    timeSpanOfRange = Math.ceil((latestTsAbsolute - earliestTsAbsolute) / 1000);
+
+    const visibleRangeMs = latestTsAbsolute - earliestTsLogs;
     if (visibleRangeMs > 0) {
```

Comment:
Could `(visibleRangeMs > 0)` be false (it's the selected range now)?

Reply:
Theoretically it can be. Cause we are (in some DS) rounding up the "to" range when sending request.  And if we receive just 1 log, it can have later time stamp than "to" range.